### ISSUE TITLE
Cant compile in XCode 10 Beta

### ISSUE THREAD

**Issue by jackjpark** [zendesk/zendesk_sdk_ios#400]
- SDK version: 2.0.3
- Xcode version: 10
- iOS or Simulator: Sim
- iOS Version: iOS 12
- Swift version: 4.1.5

Steps to Reproduce:

1. Try to compile app
2. Fails on Zendesk SDK, error: `Module compiled with Swift 4.1 cannot be imported in Swift 4.1.50`

**Comment by swillsea** [comment]
Hi @KillianSmith0, 

I'd appreciate it if your team reconsidered. Many of us have apps with Zendesk already integrated, and you're the only roadblock to us moving forward. A lack of full support is understandable, but getting the SDK to compile as a bare minimum seems like a reasonable request.

**Comment by chrisballinger** [comment]
One way to solve this problem is to define something like `USE_ZENDESK_SDK` to conditionally import it. For example in "Other Swift Flags" in only your "Release" configuration add `-DUSE_ZENDESK_SDK`. Then wherever you import or use it, put it behind `#if`:

```swift
#if USE_ZENDESK_SDK
import ZendeskSDK
#endif

func showZendesk() {
#if USE_ZENDESK_SDK
// etc
#endif
}
```

This way it's not compiled for Debug builds, but you can still use it in your Release configuration.

Alternatively you can permanently remove the Zendesk SDK and avoid companies that only provide closed source binary Swift SDKs before we've reached ABI stability.

**Comment by swillsea** [comment]
@chrisballinger true! I'll probably end up doing something similar in the meantime.

**Comment by nyeu** [comment]
Since Xcode 10 is not in beta anymore, can we have the SDK working for it?
